product_esc_code/models/product_template.py

```diff
@@ -0,0 +1,10 @@
+# Copyright 2022 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    esc_code = fields.Char()
```

Comment:
```suggestion
    alt_code = fields.Char("Alternative Code", help="Alternative product code.")
```
To make it more generic.